### PR TITLE
Add tests for file_paths_query

### PR DESCRIPTION
**Pull request by sanevova**
Followup to #4678
Adding tests for file_paths_query, fixing variable scopes, error messages when `path` is missing or empty.

### CODE REVIEW

File: osquery/config/parsers/file_paths.cpp [hunk 1]
```diff
@@ -120,10 +121,13 @@ Status FilePathsConfigParserPlugin::update(const std::string& source,
                          << "': " << sql.getMessageString();
             } else {
               for (auto& row : sql.rows()) {
-                std::string name = category.name.GetString();
                 std::string path = row["path"];
-                replaceGlobWildcards(path);
-                Config::get().addFile(source, name, path);
+                if (path == "") {
+                  LOG(ERROR) << "Cold not find non-empty `path` column in the results of file_paths_query '" << query.GetString();
```
Comment (theopolis) [inline osquery/config/parsers/file_paths.cpp]: Nitpick, having backticks in error text is weird.

File: osquery/config/parsers/file_paths.cpp [hunk 2]
```diff
@@ -120,10 +121,13 @@ Status FilePathsConfigParserPlugin::update(const std::string& source,
                          << "': " << sql.getMessageString();
             } else {
               for (auto& row : sql.rows()) {
-                std::string name = category.name.GetString();
                 std::string path = row["path"];
-                replaceGlobWildcards(path);
-                Config::get().addFile(source, name, path);
+                if (path == "") {
```
Comment (theopolis) [inline osquery/config/parsers/file_paths.cpp]: Prefer to use path.empty()

**Comment by theopolis** [comment]
ok to test

File: osquery/config/parsers/file_paths.cpp [hunk 3]
```diff
@@ -120,10 +121,15 @@ Status FilePathsConfigParserPlugin::update(const std::string& source,
                          << "': " << sql.getMessageString();
             } else {
               for (auto& row : sql.rows()) {
-                std::string name = category.name.GetString();
                 std::string path = row["path"];
-                replaceGlobWildcards(path);
-                Config::get().addFile(source, name, path);
+                if (path.empty()) {
```
Comment (akindyakov) [inline osquery/config/parsers/file_paths.cpp]: There is nit. Speaking semantically, this loop just need to read rows and fetch all paths from it. That means there is no reason for row to be non-const. Also, `operator[]` for the `std::map` is non-const and it creates default value in case of key absense. Which is not perfect.

You could de it with iterator:
```c++
for (const auto& row : sql.rows()) {
  auto pathIt = row.find("path");
  if (pathIt == row.end()) {
     LOG(ERROR) << ...
  } else {
     auto path = *pathIt;
     ...
  }
```
